### PR TITLE
Added integration with Amazon RDS

### PR DESCRIPTION
**Pull request by maksym-lebedynets**
All criteria are done, including additional.

Links for endpoints:
GET ALL
https://yp0ekii2cc.execute-api.eu-west-1.amazonaws.com/dev/products
GET ONE
https://yp0ekii2cc.execute-api.eu-west-1.amazonaws.com/dev/products/cd25dbd1-f7a7-488a-8ddb-e9822e8443f5
CREATE PRODUCT:
https://yp0ekii2cc.execute-api.eu-west-1.amazonaws.com/dev/products/
POST request with body:
{
    "title": "iPod Nano",
    "description": "player",
    "price": 500
}
After creating a new product, you can check it with GET ALL endpoint.

Link for FE PR:
https://github.com/maksym-lebedynets/aws-nodejs-program/pull/3

Link to FE app: https://d2o1n28rymxg96.cloudfront.net/

Additional criteria:
- validation for Product creation on API Gateway side
- handle unexpected error for all lambdas
- console.log for all lambdas
- transaction for `createProduct` lambda 

Validation schema screenshot:
Снимок экрана 2020-11-07 в 13 58 16

### CODE REVIEW

**Comment by shohrukh92** [comment]
Validation of the product should return 400 error in case of wrong data format
image

**Comment by maksym-lebedynets** [comment]
> Validation of the product should return 400 error in case of wrong data format
> image

Could you check it again?

File: product-service/createProduct.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    await client.query('BEGIN');
+
+    const { title, description, price } = JSON.parse(event.body);
+    const addProductQuery = 'insert into products (title, description, price) values ($1, $2, $3) returning *';
+    const { rows } = await client.query(addProductQuery, [title, description, price]);
+    const addToStockQuery = 'insert into stocks(product_id, count) values ($1, $2)';
+    const productId = rows[0].id;
+
+    await client.query(addToStockQuery, [productId, 1]);
+    await client.query('COMMIT');
```
Comment (IvanPresmytsky) [inline product-service/createProduct.js:17]: Would be nice to move all db-specific logic outside the lamdas.

**Review by IvanPresmytsky — APPROVED** [review]
Good job!

**Comment by IvanPresmytsky** [comment]
> > Validation of the product should return 400 error in case of wrong data format
> > image
> 
> Could you check it again?

It works fine for me.

File: product-service/createProduct.js (hunk 1 above)
Comment (maksym-lebedynets) [reply product-service/createProduct.js:17]: Good point

**Comment by mousemckill** [comment]
image

**Comment by maksym-lebedynets** [comment]
What do you want to check? You did GET request with specified body, it is impossible

Скачайте Outlook для Android<https://aka.ms/ghei36>
________________________________

**Comment by ihargit** [comment]
image
Hi, as I understand, count is always 1. Might be fixed)

**Comment by ihargit** [comment]
image
Also, description accepts number.

**Comment by maksym-lebedynets** [comment]
> image
> Hi, as I understand, count is always 1. Might be fixed)

Oops, you are right :) I'll fix

**Comment by ihargit** [comment]
Please, next time provide link to your FE app

**Comment by maksym-lebedynets** [comment]
> image
> Hi, as I understand, count is always 1. Might be fixed)

Idea is to create only product, the system by its own set count to 1 by default.

**Comment by maksym-lebedynets** [comment]
> Please, next time provide link to your FE app

Link added

**Pull request merged**